linera-core/src/client/mod.rs

```diff
@@ -1526,6 +1588,29 @@ where
         }
     }
 
+    /// Synchronizes all chains that any application on this chain subscribes to.
+    async fn synchronize_publisher_chains(&self) -> Result<(), ChainClientError> {
+        let chain_ids = self
+            .chain_state_view()
+            .await?
+            .execution_state
+            .system
+            .event_subscriptions
+            .indices()
+            .await?
+            .iter()
+            .map(|(chain_id, _)| *chain_id)
+            .filter(|chain_id| *chain_id != self.chain_id && *chain_id != self.admin_id)
```

Comment:
Shouldn't we synchronize the admin chain always?

Reply:
Yes, but we already do that in `synchronize_from_validators`, so we avoid doing it again here. I'll add a comment.

Reply:
It'd be nice if we didn't have to think about this. Maybe a separate abstraction (struct) responsible for synchronizing other chains that holds the necessary information (like tracked chains) and a simple API that we can use to trigger the sync?

Reply:
Not sure what's the best abstraction here. This function is only called in one place, in `synchronize_from_validators`.

But actually, maybe we don't have to synchronize the admin chain there explicitly, and can just always consider the admin chain to be a publishing chain…

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/3784/commits/50d0eaee76f7125aad57f0d4a5e9a0e06c76d7c5.